src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -155,6 +155,40 @@ public IActionResult DrainStatus([FromServices] IScriptHostManager scriptHostMan
             }
         }
 
+        [HttpPost]
+        [Route("admin/host/resume")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        public async Task<IActionResult> Resume([FromServices] IScriptHostManager scriptHostManager)
+        {
+            _logger.LogDebug($"Received request to resume a draining host.");
+
+            if (Utility.TryGetHostService(scriptHostManager, out IDrainModeManager drainModeManager))
+            {
+                _logger.LogDebug($"Host status: {scriptHostManager.State.ToString()}. Drain mode enabled: {drainModeManager.IsDrainModeEnabled}");
+
+                if (scriptHostManager.State != ScriptHostState.Running)
+                {
+                    _logger.LogDebug("Host is not running, cannot resume");
+                    return StatusCode(StatusCodes.Status409Conflict);
+                }
+
+                if (!drainModeManager.IsDrainModeEnabled)
+                {
+                    _logger.LogWarning("Resume request received when not in drain mode");
+                    return Ok(new { hostStatus = scriptHostManager.State.ToString() });
+                }
+
+                _logger.LogDebug("Host is running and in drain mode, starting a new host");
+                await scriptHostManager.RestartHostAsync();
```

Comment:
Do we know how long this method call takes? I wonder we will run into a situation where the request from the caller(scale controller?) times out because this `RestartHostAsync` method takes a longer time to finish.